src/process/dust/ccpr_dust_common_mod.F90

```diff
@@ -0,0 +1,405 @@
+      REAL(fp), ALLOCATABLE           :: UpperBinRadius(:)         ! Upper bin radius        [m]
+      REAL(fp), ALLOCATABLE           :: EffectiveRadius(:)        ! Effective radius        [m]
+      REAL(fp), ALLOCATABLE           :: DustDensity(:)            ! Dust density            [kg/m^3]
+      REAL(fp)                        :: BetaScaleFactor           ! Beta Scaling Parameter  [1]
+      REAL(fp)                        :: AlphaScaleFactor          ! Alpha Scaling Parameter [1]
+      REAL(fp), ALLOCATABLE           :: TotalEmission             ! Total emission          [kg/m^2/s]
+      REAL(fp), ALLOCATABLE           :: EmissionPerSpecies(:)     ! Emission per species    [kg/m^2/s]
+
+      ! Scheme Options
+      INTEGER                         :: MoistOpt  ! Fengsha-Moisture Calculation Option
+      INTEGER                         :: DragOpt      ! Fengsha-Drag Calculation Option
+      INTEGER                         :: HorizFluxOpt        ! Horizontal Flux Calculation Option
+
+      !=================================================================
+      ! Module specific variables/arrays/data pointers come below
+      !=================================================================
+
+   END TYPE DustStateType
+
+contains
+   !>
+   !! \brief Computes the soil moisture attenuation factor for dust emission
+   !!
+   !! Fecan, F., Marticorena, B., and Bergametti, G.: Parametrization of the increase of the aeolian
+   !! erosion threshold wind friction velocity due to soil moisture for arid and semi-arid areas,
+   !! Ann. Geophys., 17, 149–157, https://doi.org/10.1007/s00585-999-0149-7, 1999.
+   !!
+   !! \param clay Fractional clay content
+   !! \param sand Fractional sand content
+   !! \param volumetric_soil_moisture Volumetric soil moisture
+   !! \param H Soil moisture attenuation factor for dust emission
+   !!
+   !!!>
+   subroutine Fecan_SoilMoisture( clay, sand, volumetric_soil_moisture, H)
+      IMPLICIT NONE
+      ! Parameters
+      !-----------
+      real(fp), intent(in)  :: clay                      ! Fractional Clay Content
+      real(fp), intent(in)  :: sand                      ! Fractional Sand Content
+      real(fp), intent(in)  :: volumetric_soil_moisture  ! Volumetric soil moisture [m3 m-3]
+      real(fp), intent(out) :: H                         ! Soil Moisture attenuation factor for dust emission [1]
+
+      ! Local Variables
+      !----------------
+      real(fp) :: vsat                      ! Saturated volumetric water content (sand-dependent) ! [m3 m-3]
+      real(fp) :: gravimetric_soil_moisture ! Gravimetric soil moisture [kg/kg]
+      real(fp) :: DryLimit                  ! Dry limit of the soil moisture [kg/kg]
+
+      ! Initialize
+      !-----------
+      H = ZERO
+
+      !--------------------------------------------
+      ! Compute Saturated Volumetric Water Content
+      !--------------------------------------------
+      vsat = 0.489 - 0.00126 * (100. * sand)
```

Comment:
```suggestion
      vsat = 0.489_fp - 0.00126_fp * (100._fp * sand)
```
To ensure desired accuracy, all float literals should get `_fp`, otherwise they will get the compiler default for `real`, probably single precision.